From: Hannes Reinecke <hare@suse.de>
Date: Tue, 25 Jun 2013 15:12:04 +0200
Subject: scsi: Retry failfast commands after EH
References: bnc#798050
Patch-Mainline: Submitted, linux-scsi Jan 27 2014

Any commands with return status DID_FAILFAST_XXX should be
retried; the port will be in a sane state after EH.

Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/scsi_error.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/scsi/scsi_error.c b/drivers/scsi/scsi_error.c
index 01adc4f..3fa1cab 100644
--- a/drivers/scsi/scsi_error.c
+++ b/drivers/scsi/scsi_error.c
@@ -534,6 +534,9 @@ static int scsi_eh_completed_normally(struct scsi_cmnd *scmd)
 		 */
 		return scsi_check_sense(scmd);
 	}
+	if ((host_byte(scmd->result) == DID_TRANSPORT_DISRUPTED) ||
+	    (host_byte(scmd->result) == DID_TRANSPORT_FAILFAST))
+		return NEEDS_RETRY;
 	if (host_byte(scmd->result) != DID_OK)
 		return FAILED;
 
-- 
1.7.12.4

